specs/Cryptography/Types.md

```diff
@@ -0,0 +1,72 @@
+
+
+# signature.rs
+
+PublicKey/SecretKey types are used to represent keypairs for Ed25519 or Secp256k1.
+
+```rust
+pub enum PublicKey {
+    ED25519(ED25519PublicKey),
+    SECP256K1(Secp256K1PublicKey),
+}
+```
+```rust
+pub enum SecretKey {
+    ED25519(ED25519SecretKey),
+    SECP256K1(secp256k1::key::SecretKey),
+}
+```
+```rust
+pub enum Signature {
+    ED25519(ed25519_dalek::Signature),
+    SECP256K1(Secp256K1Signature),
+}
+```
+
+The inner types `ED25519PublicKey`, `ED25519SecretKey`, `Secp256K1PublicKey`, `secp256k1::key::SecretKey`
+are byte arrays of 32 or 64 bytes which are interpreted as described in sections [Ed25519](Ed25519.md)/[Secp256k1](Secp256k1.md).
+
+## JSON Serialization
+
+In string/JSON format keys/signatures are written as the type (`ed25519`/`secp256k1`) followed by a
+semicolon and the underlying data, which is represented as a byte array (as defined by the corresponding
+standard) encoded as `base58`.
```

Comment:
What standard? Either we should precisely reference a standard that describes a serialization format, or we should include the description in the specification.